client/state/data-layer/wpcom-api-middleware.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * External dependencies
+ */
+import { has } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import handlers from './wpcom';
+
+export const middleware = store => next => action =>
+	has( handlers, action.type )
+		? handlers[ action.type ].forEach( handler => handler( store, action ) )
```

Comment:
performance-wise this won't matter and I'm not sure we want to prevent use of `getState()`. this is actually _the place_ to perform side-effecting transactions, _the place_ to mangle related data.